turbopack/crates/turbopack-resolve/src/resolve.rs

```diff
@@ -196,6 +196,23 @@ async fn base_resolve_options(
         environment.resolve_extensions().owned().await?
     } else {
         let mut ext = Vec::new();
+        
+        if opt.enable_react_native_web_infix {
+            if opt.enable_typescript && opt.enable_react {
+                ext.push(rcstr!(".web.tsx"));
+            }
+            if opt.enable_typescript {
+                ext.push(rcstr!(".web.ts"));
+            }
+            if opt.enable_react {
+                ext.push(rcstr!(".web.jsx"));
+            }
+            ext.push(rcstr!(".web.js"));
+            if opt.enable_mjs_extension {
+                ext.push(rcstr!(".web.mjs"));
+            }
+        }
+        
```

Comment:
maybe instead, after `ext` is created, do
```rust
if opt.enable_react_native_web_infix {
	ext = ext.iter().map(|e| format!(".web{e}").into()).chain(ext.into_iter()).collect();
}
```
i.e. take the list, prepend `.web` to everything, and add the original extension list at the end.